add_up.rb

```diff
@@ -0,0 +1,19 @@
+## recursive
+# def add_up(i)
+#   if i > 0
+#     i + add_up(i-1)
+#   else
+#     0
+#   end
+# end
+
+## iterator
+def add_up(i)
+  sum = 0
+  i.downto(1) { |i| sum += i}
```

Comment:
This is cool! I didn't know about `.downto`. It works well here.

An alternative would be to do a range and Ruby has a helper method to sum

```rb
def add_up(i)
  (1..i).sum
end
```

Reply:
It appeared I didn't study range enough. ;-)